c++/src/kj/common-test.c++

```diff
@@ -818,6 +818,188 @@ KJ_TEST("ArrayPtr operator ==") {
   KJ_EXPECT(ArrayPtr<double>(d, 1) != ArrayPtr<double>(d, 1));
 }
 
+KJ_TEST("ArrayPtr operator <=>") {
+  KJ_EXPECT((ArrayPtr<const int>({}) <=> nullptr) == std::strong_ordering::equal);
```

Comment:
wow, this is wall of text. Can we do something better here?

Couple things come to mind:
- local typedefs
- data-driven tests

Reply:
I think the latest updates have taken care of this, I'd love to see if there are ways to make it even more clean.